lib/cluster_ht_item.rb

```diff
@@ -67,19 +67,25 @@ def update_or_add_ht_items(cluster)
     Services.logger.debug "Cluster #{cluster.inspect}: " \
       "adding ht_items #{ht_items.inspect} with ocns #{@ocns}"
     to_append = []
+    needs_reclustering = false
     ht_items.each do |item|
       if (existing_item = cluster.ht_item(item.item_id))
         Services.logger.debug "updating existing item with id #{item.item_id}"
+        needs_reclustering = true unless current_is_subset_of_new(existing_item.ocns, item.ocns)
         existing_item.update_attributes(item.to_hash)
       else
         ClusterHtItem.new(item).delete
         to_append << item
       end
     end
 
-    unless to_append.empty?
-      cluster.add_ht_items(to_append)
-    end
+    cluster.add_ht_items(to_append) unless to_append.empty?
+
+    Reclusterer.new(cluster).recluster if needs_reclustering
+  end
+
+  def current_is_subset_of_new(current_ocns, new_ocns)
```

Comment:
Won't this also, unnecessarily, trigger if current_ocns == new_ocns?
(bacause [1,2] is a subset of [1,2])

Reply:
needs_reclustering is false unless current_is_subset_of_new